crates/astria-conductor/src/execution_client.rs

```diff
@@ -1,88 +1,128 @@
-use astria_proto::generated::execution::v1alpha1::{
+use astria_proto::generated::execution::v1alpha2::{
     execution_service_client::ExecutionServiceClient,
-    DoBlockRequest,
-    DoBlockResponse,
-    FinalizeBlockRequest,
-    InitStateRequest,
-    InitStateResponse,
+    BatchGetBlocksRequest,
+    BatchGetBlocksResponse,
+    Block,
+    BlockIdentifier,
+    CommitmentState,
+    ExecuteBlockRequest,
+    GetBlockRequest,
+    GetCommitmentStateRequest,
+    UpdateCommitmentStateRequest,
 };
 use color_eyre::eyre::Result;
 use prost_types::Timestamp;
 use tonic::transport::Channel;
 use tracing::info;
 
+/// Creates a new RPC Client for Execution
+///
+/// # Arguments
+///
+/// * `address` - The address of the RPC server that we want to communicate with.
+pub(crate) async fn new(address: &str) -> Result<ExecutionServiceClient<Channel>> {
+    let client = ExecutionServiceClient::connect(address.to_owned()).await?;
+    info!("Connected to execution service at {}", address);
+    Ok(client)
+}
+
 #[async_trait::async_trait]
 pub(crate) trait ExecutionClient {
```

Comment:
Small nit: might call this `ExecutionClientExt` to make it clear this is an extension trait, but no big deal. Many other crates in the ecosystem would also just call this `Client` without `Ext`.